crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -0,0 +1,281 @@
+			"Plain text chain specification file generated at: {}",
+			plain_chain_spec.display()
+		));
+
+		// Customize spec based on input.
+		let relay = self.relay.unwrap_or(RelayChain::PaseoLocal).into_str();
+		let _ = replace_relay_spec(&plain_chain_spec, relay, "rococo-local");
+		let chain_type = self.chain_type.unwrap_or(ChainType::Development).into_str();
+		let _ = replace_chain_type(&plain_chain_spec, chain_type, "Local");
+
+		let raw_chain_spec =
+			generate_raw_chain_spec(&binary_path, &plain_chain_spec, RAW_CHAIN_SPEC_FILE_NAME)?;
+		generated_files.push(format!(
+			"Raw chain specification file generated at: {}",
+			raw_chain_spec.display()
+		));
+		let wasm_file_name = format!("para-{}-wasm.wasm", para_id);
+		let wasm_file = export_wasm_file(&binary_path, &raw_chain_spec, &wasm_file_name)?;
+		generated_files.push(format!(
+			"WebAssembly runtime file exported at: {}",
+			wasm_file.display().to_string()
+		));
+		let genesis_file_name = format!("para-{}-genesis-state", para_id);
+		let genesis_state_file =
+			generate_genesis_state_file(&binary_path, &raw_chain_spec, &genesis_file_name)?;
+		generated_files.push(format!(
+			"Genesis State exported at {} file",
+			genesis_state_file.display().to_string()
+		));
+
+		console::Term::stderr().clear_last_lines(5)?;
+		let generated_files: Vec<_> = generated_files
+			.iter()
+			.map(|s| style(format!("{} {s}", console::Emoji("●", ">"))).dim().to_string())
+			.collect();
+		cli.success(format!("Generated files:\n{}", generated_files.join("\n")))?;
+		cli.outro(format!(
+			"Need help? Learn more at {}\n",
+			style("https://learn.onpop.io").magenta().underlined()
+		))?;
+
+		Ok("spec")
+	}
+}
+
+/// Guide the user to generate thier chain specification.
+async fn guide_user_to_generate_spec() -> anyhow::Result<BuildSpecCommand> {
+	Cli.intro("Generate your chain spec")?;
+
+	// Confirm project path
+	let target_path: String = input("Where is the project located?")
+		.placeholder("./")
+		.default_input("./")
+		.interact()?;
+
+	// Prompt for chain id.
+	let para_id: u32 = input("What parachain ID should the build use?")
+		.placeholder("2000")
+		.default_input("2000")
+		.interact()?;
+
+	// Prompt for relay chain.
+	let mut prompt =
+		cliclack::select("Choose the relay chain your chain will be connecting to: ".to_string());
+	for (i, relay) in RelayChain::VARIANTS.iter().enumerate() {
+		if i == 0 {
+			prompt = prompt.initial_value(relay);
+		}
+		prompt = prompt.item(relay, relay.clone().into_str(), "");
+	}
+	let rc = prompt.interact()?;
+	let relay_chain = rc.clone();
+
+	// Prompt for chain type.
+	let mut prompt = cliclack::select("Choose the chain type: ".to_string());
+	for (i, chain_type) in ChainType::VARIANTS.iter().enumerate() {
+		if i == 0 {
+			prompt = prompt.initial_value(chain_type);
+		}
+		prompt = prompt.item(chain_type, chain_type.clone().into_str(), "");
+	}
+	let ct = prompt.interact()?;
+	let chain_type = ct.clone();
```

Comment:
Just a question here, is the `clone` the best way to get the value?

Reply:
The execution of the command requires a the type `RelayChain` or `ChainType` respectively, the interaction method is returning a reference.

Used clone to get me going quickly, but I'll see if I can get this implemented in a cleaner way, didn't love it when I first write it, but at least allowed me to continue.